### PR TITLE
feat(optimizer): add qwikVite configuration option optimizerOptions.inlineStylesUpToBytes

### LINKED ISSUE #5069: Optimizer inlines large blobs of styles
The optimizer inlines css below a certain length/size, which is awesome for "small" stuff. I use tailwind and it adds quite a bit, but apparently not enough :)

https://github.com/BuilderIO/qwik/blob/86a9452be1572c84868a5ebee3ad417cf304b70f/packages/qwik/src/optimizer/src/plugins/vite.ts#L463

Changing the setting from 20000 to 200 makes my global.css a separate file again.
-rw-r--r--@ 1 roel  wheel  26671 Aug 31 12:17 /tmp/2e2.html
-rw-r--r--@ 1 roel  wheel  40947 Aug 31 12:17 /tmp/2e4.html

I personally could try to not use global.css/tailwind, maybe use pandacss instead. Maybe make the 20000 configurable?

Any other opinions?

### PR DESCRIPTION
**Pull request by jaroel**
Closes #5069

# What is it?

- [X] Feature / enhancement
- [ ] Bug
- [ ] Docs / tests / types / typos

# Description

The Optimizer inlines styles up to 2e4 bytes. This PR makes that value configurable using:
```
qwikVite({
  optimizerOptions: {
    inlineStylesUpToBytes: 400,
  },
}),
```

Feedback needed:
1. is `inlineStylesUpToBytes` a good enough name?
2. is placing it inside `optimizerOptions` the correct scope?

# Use cases and why

- 1. Inlining 20k bytes of css might be invalid for certain use cases.

# Checklist:

- [X] My code follows the [developer guidelines of this project](https://github.com/BuilderIO/qwik/blob/main/CONTRIBUTING.md)
- [X] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] Added new tests to cover the fix / functionality

### CODE REVIEW

**Comment by jaroel** [comment]
This works. Could use some docs and some tests.

File: packages/qwik/src/optimizer/src/plugins/plugin.ts [hunk 1]
```diff
@@ -266,6 +267,11 @@ export function createPlugin(optimizerOptions: OptimizerOptions = {}) {
     }
     opts.csr = !!updatedOpts.csr;
 
+    opts.inlineStylesUpToBytes = optimizerOptions.inlineStylesUpToBytes ?? 0;
```
Comment (jaroel) [inline packages/qwik/src/optimizer/src/plugins/plugin.ts:270]: We should add a default value here.

Comment (jaroel) [reply packages/qwik/src/optimizer/src/plugins/plugin.ts:270]: Maybe having the _option_ to inline is worth more than inlining by default. Especially when an external .css file can be hashed and cached forever.

Comment (mhevery) [reply packages/qwik/src/optimizer/src/plugins/plugin.ts:270]: I think we should keep the previous behavior unless it is overwritten...

**Review by mhevery — CHANGES_REQUESTED** [review]

**Review by mhevery — APPROVED** [review]
Added my changes

**Comment by mhevery** [comment]
Thank you for your contribution ❤️

**Comment by jaroel** [comment]
Thanks for the fixup, nice touch with the docstring!

**Pull request merged**